### DISCUSSION TITLE
Is there a way to delete metrics in-memory

### DISCUSSION THREAD

**Discussion by snguyen64** [open-telemetry/opentelemetry-go#3402] [Q&A] [answered]
Coming from prometheus, you can create MetricVec type which can delete specific metrics.

I went through the opentelemetry sdk code but didn't see any indication of a way for users to delete metrics manually. I've only seen that the in-memory metrics are only deleted through delta temporality.

Is there a way to delete in memory metrics similar to how prometheus does this? If not, will this be in the roadmap?

**Comment by dmathieu** [comment, marked as answer]
I'm not even seeing that mentioned in the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md
If this were to be added, it would have to be specified first.

**Reply by MrAlias** [reply]
Agreed :+1: 

For what it's worth, there is [this issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1891) in the specification, and [this issue](https://github.com/open-telemetry/opentelemetry-go/issues/3006) here. The specification issue relates to this in that is is proposing standardizing bounding existing unlimited memory use, and the issue in this repository is tracking work we will need once that specification issue is defined.

There isn't however any issue tracking user specified memory limiting as far as I know.

**Reply by asafm** [reply]
I couldn't find such an issue myself, so I created one: https://github.com/open-telemetry/opentelemetry-specification/issues/3062
